### PR TITLE
(maint) Update developer docs

### PR DESCRIPTION
**Pull request by lucywyman**
**What this changes** Include more detailed documentation on testing with local containers for bolt-server, including how to get the IP of the server making the request so that it is appropriately routed to the local targets.
**Why** I made some dumb mistakes while setting up testing, and want to save others the trouble.

### CODE REVIEW

File: developer-docs/bolt-server.md [hunk 1]
```diff
@@ -229,52 +224,62 @@ curl -v --cacert $BOLT_CACERT --cert $BOLT_CERT --key $BOLT_KEY $BOLT_ROOT
...
+  "host-key-check":false},
+"parameters":{
+  "message":"Hello World"}}
 ```
-You should then be able to post it with:
+**Verify that the target information** is correct, and change it if you want to use a different target. You can find other example requests in the `examples` directory.
 
+You should then be able to post it with:
 ```
 curl -X POST -H "Content-Type: application/json" -d @request.json --cacert $BOLT_CACERT --cert $BOLT_CERT --key $BOLT_KEY $BOLT_ROOT $BOLT_ROOT/ssh/run_task
```
Comment (donoghuc) [inline developer-docs/bolt-server.md]: Extra `$BOLT_ROOT` there. Should be `curl -X POST -H ation/json" -d @request.json --cacert $BOLT_CACERT --cert $BOLT_CERT --key $BOLT_KEY  $BOLT_ROOT/ssh/run_task`. Also i think it is important to show the full output 
```
{"node":"172.18.0.1","status":"success","result":{"_output":"ac80223bd3b4 got passed the message: hey\n"}}
```
The `node` value might be confusing bc it is not running on the docker host (your laptop) instead it is running against the bolt-spec container. Showing the `_output` is helpful as it should be the container ID the task is run against
```
ac80223bd3b4        bolt-spec           "/usr/sbin/sshd -D"      30 minutes ago      Up 30 minutes               0.0.0.0:20022->22/tcp      ubuntu_node
```

File: developer-docs/bolt-server.md [hunk 2]
```diff
@@ -229,52 +224,62 @@ curl -v --cacert $BOLT_CACERT --cert $BOLT_CERT --key $BOLT_KEY $BOLT_ROOT
...
 
 ```
-bundle exec scripts/server_client.rb sample::echo ssh '{"message": "hey"}'
+bundle exec scripts/server_client.rb sample::echo <TARGET> '{"message": "hey"}'
+```
+Where `<TARGET>` is either:
+* A vmpooler VM. To use this, replace `<TARGET>` above with the hostname.
+* One of the containers brought up by the `docker-compose` in the `spec` directory. To use these, you'll want to get the IP of the container bolt-server is running on:
+```
+docker exec -it bolt_boltserver_1 /sbin/ip route|awk '/default/ { print $3 }'
```
[the remark this answers is not in the document]
Comment (adreyer) [reply developer-docs/bolt-server.md]: The command you document doesn't work for me when I try to run it against one of the containers we set up for spec testing.

File: developer-docs/bolt-server.md [hunk 3]
```diff
@@ -229,52 +224,62 @@ curl -v --cacert $BOLT_CACERT --cert $BOLT_CERT --key $BOLT_KEY $BOLT_ROOT
...
+environment so run `docker-compose up -d --build` from the spec directory
+before trying to use it.
 
 ```
-bundle exec scripts/server_client.rb sample::echo ssh '{"message": "hey"}'
+bundle exec scripts/server_client.rb sample::echo <TARGET> '{"message": "hey"}'
+```
+Where `<TARGET>` is either:
+* A vmpooler VM. To use this, replace `<TARGET>` above with the hostname.
+* One of the containers brought up by the `docker-compose` in the `spec` directory. To use these, you'll want to get the IP of the container bolt-server is running on:
```
Comment (donoghuc) [inline developer-docs/bolt-server.md]: I think it is worth pointing out that any task in `bolt/spec/fixtures/modules/sample` will exist in the `puppetserver` image.  For example you could run the task defined with `init.sh` with `bundle exec scripts/server_client.rb sample bolt:bolt@172.18.0.1:20022  '{"message": "hey"}' ` 
with expected output:
```
{
  "node": "172.18.0.1",
  "status": "success",
  "result": {
    "_output": "hey\n"
  }
}
```

Comment (adreyer) [reply developer-docs/bolt-server.md]: `bolt/spec/fixtures/modules`

File: developer-docs/bolt-server.md [hunk 4]
```diff
@@ -229,52 +224,66 @@ curl -v --cacert $BOLT_CACERT --cert $BOLT_CERT --key $BOLT_KEY $BOLT_ROOT
...
+"parameters":{
+  "message":"hey"}}
 ```
-You should then be able to post it with:
+**Verify that the target information** is correct, and change it if you want to use a different target. You can find other example requests in the `examples` directory.
 
+You should then be able to post it with:
 ```
-curl -X POST -H "Content-Type: application/json" -d @request.json --cacert $BOLT_CACERT --cert $BOLT_CERT --key $BOLT_KEY $BOLT_ROOT $BOLT_ROOT/ssh/run_task
+curl -X POST -H "Content-Type: application/json" -d @developer-docs/examples/ssh-echo.json --cacert $BOLT_CACERT --cert $BOLT_CERT --key $BOLT_KEY $BOLT_ROOT/ssh/run_task
```
Comment (donoghuc) [inline developer-docs/bolt-server.md]: Good idea!

File: developer-docs/bolt-server.md (hunk 3 above)
Comment (donoghuc) [reply developer-docs/bolt-server.md]: Right. I guess we could just link to https://github.com/puppetlabs/bolt/blob/d0473f31ba9fefdf715e7ab7b76aeef05ab91f3d/spec/docker-compose.yml#L29

**Pull request merged**